pkg/genericapiserver/genericapiserver.go

```diff
@@ -269,6 +269,10 @@ type GenericAPIServer struct {
 	enableOpenAPISupport   bool
 	openAPIInfo            spec.Info
 	openAPIDefaultResponse spec.Response
+
+	// PostStartHooks is called in a go func after the server has started listening.
+	// It may kill the process with a panic if it wishes to by returning an error
+	PostStartHooks []func() error
```

Comment:
> For debugability, funcs need a name, so we can say what went wrong when they fail.
> Funcs should have a named type.

Sure.



> They should accept some sort of PostStartContext type which is a struct, so we can add fields to it without changing the entire codebase

I'm less sure of this.  How about a client/clientconfig?  I see this as a structured way to optimize instead of a separate binary that does an initialization step as a separate call after API server start.